savaslabs/savaslabs.github.io
# Issue #3392: Implement spelling, prose, and markdown checkers in pre-commit hook

Fixes issue [#3392](https://pm.savaslabs.com/issues/3392)

## Summary of changes

- Adds a pre-commit hook to test:
  - Spelling via Aspell
  - Markdown via markdown-lint
  - Prose via proselint
- Updates the README to add installation and usage instructions
- Adds a pull request template
- Corrects some spelling mistakes I found in blog posts during testing

## Notes

To be discussed at Skillshare on 2017-08-09!

## To test

- Pull this branch
- Run `git config core.hooksPath hooks` to configure the git hook directory
- Run `_scripts/aspell.sh` in the root of the repository. This will install and configure Aspell. **Let me know if you have any issues here**
- Edit a markdown file, such as a blog post or the README. Misspell a word or two and add a few lines of blank space.
- `git add .` to stage the file
- `git commit`
- You'll get some output:
  - First from Aspell, you'll see the words you misspelled, and probably some others that are proper nouns or development terms/commands/code
  - Then you should get an error from mdl (markdown lint) about the extra blank lines you added
  - Finally there will likely be some proselint suggestions
- Edit the `savas_wordlist.txt` file and add one of the misspelled words to the list (a proper noun or development term, not an actual misspelled word please). Run `_scripts/aspell.sh` again.
- Run `git commit` again. The word you added to the word list should no longer appear in the list of misspelled words.
- Run `git commit --no-verify` to commit without fixing all the tests

### Pull request reviewer

As the reviewer, I have verified the following:
